kafka-client/src/main/java/dev/responsive/kafka/internal/db/mongo/MongoSessionTable.java

```diff
@@ -380,18 +381,24 @@ public KeyValueIterator<SessionKey, byte[]> fetchAll(
     );
 
     final var minKey = new SessionKey(key, 0, earliestSessionEnd);
-    final var maxKey = new SessionKey(key, 0, latestSessionEnd);
+    final var maxKey = new SessionKey(key, Long.MAX_VALUE, latestSessionEnd);
```

Comment:
This is the bug, I assume...I figured it would be something silly like this. Sorry for not catching it during review 

But totally awesome that you caught this in the test. You've already picked up on so many testing gaps and started to improve the testing framework 🙏